src/UserForm.test.js

```diff
@@ -0,0 +1,124 @@
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { REQUEST_FORM_FIELD_NAMES, requestStatuses } from './constants';
+import UserForm from './UserForm';
+
+jest.mock('./utils', () => ({
+  ...jest.requireActual('./utils'),
+  getFullName: jest.fn(() => ({ user: '12', proxy:'11' })),
+  userHighlightBox: jest.fn(() => 'userHighlightBox'),
+}));
+
+const user = {
+  id: REQUEST_FORM_FIELD_NAMES.REQUESTER_ID,
+  firstName: 'Jenny',
+  lastName: 'Smith',
+  barcode: REQUEST_FORM_FIELD_NAMES.REQUESTER_BARCODE,
+};
+const deliveryLocations = [
+  { value: '1', label: REQUEST_FORM_FIELD_NAMES.DELIVERY_ADDRESS_TYPE_ID },
+];
+const fulfilmentTypeOptions = [
+  { labelTranslationPath: 'ui-requests.fulfilmentTypes.delivery', value: 'delivery' },
+  { labelTranslationPath: 'ui-requests.fulfilmentTypes.pickup', value: 'pickup' },
+];
+const servicePoints = [
+  { id: '1', name: REQUEST_FORM_FIELD_NAMES.PICKUP_SERVICE_POINT_ID },
+];
+const onChangeAddress = jest.fn();
+const onChangeFulfilment = jest.fn();
+const onCloseProxy = jest.fn();
+const onSelectProxy = jest.fn();
+
+const formProps = {
+  onChangeAddress,
+  onChangeFulfilment,
+  onCloseProxy,
+  onSelectProxy,
+  deliveryLocations,
+  fulfilmentTypeOptions,
+  servicePoints,
+  stripes: { connect: jest.fn((component) => component) },
+};
+describe('UserForm', () => {
+  it('should render delivery address select when deliverySelected is true', () => {
+    const defaultProps = {
+      deliverySelected: true,
+      user,
+      patronGroup: 'Staff',
+      proxy: { id: REQUEST_FORM_FIELD_NAMES.PROXY_USER_ID, firstName: 'Anna', lastName: 'Grey', barcode: '123456789' },
+      optionLabel: 'testoptionLabel',
+      request: { status : requestStatuses.AWAITING_DELIVERY }
+    };
+    render(<UserForm {...defaultProps} {...formProps} />);
+    userEvent.selectOptions(
+      screen.getByRole('combobox', { name: 'ui-requests.deliveryAddress' }),
+      screen.getByRole('option', { name: 'deliveryAddressTypeId' }),
+    );
+    expect(screen.getByRole('option', { name: 'deliveryAddressTypeId' }).selected).toBe(true);
+  });
+  it('should render servicePoints when deliverySelected is false', () => {
+    const props = {
+      formProps,
+      deliverySelected: false,
+      user,
+      patronGroup: 'Staff',
+      proxy: { id: REQUEST_FORM_FIELD_NAMES.PROXY_USER_ID, firstName: 'Anna', lastName: 'Grey', barcode: '123456789' },
+      optionLabel : 'testoptionLabel',
+      request: { status : requestStatuses.AWAITING_DELIVERY },
+    };
+    render(<UserForm {...props} {...formProps} />);
+    userEvent.selectOptions(
+      screen.getByRole('combobox', { name: 'ui-requests.pickupServicePoint.name' }),
+      screen.getByRole('option', { name: 'pickupServicePointId' }),
+    );
+    expect(screen.getByRole('option', { name: 'pickupServicePointId' }).selected).toBe(true);
+  });
+  it('should render UserForm', () => {
```

Comment:
I am not sure that we need that test.

@Dmitriy-Litvinenko what do you think about it?